numpy/testing/utils.py

```diff
@@ -343,16 +343,27 @@ def assert_equal(actual,desired,err_msg='',verbose=True):
         except AssertionError:
             raise AssertionError(msg)
 
+    def isnat(x):
+        return (hasattr(x, 'dtype')
+                and getattr(x.dtype, 'kind', None) in 'mM'
```

Comment:
`None` is dangerous, it will raise an error with the string `'mM'`. Using another string would work, say `'_'`.